content/articles/creating-applet-with-threads/index.md

```diff
@@ -0,0 +1,156 @@
+    ...
+}
+```
+
+You need to change the signature of your applet class as shown below:
+
+```java
+public class AppletClass extends java.applet.Applet implements Runnable
+{
+    ...
+}
+```
+
+The program class above supports the runnable interface. Remember that interfaces are a mechanism to add method names that are shared by many classes.  They can then be combined and implemented within different classes that need to implement `behavior`. 
+
+The `run()` method is identified by the runnable interface then we add a `start()` function. For example:
+
+```java
+public start()
+ {
+ if (runner == null);
+ {
+  runner = new Thread(this);
+  runner.start();
+ }
+ }
+```
+
+Once users modify the `start()` method, it does nothing but regenerate a thread. The applet's body is switched to a new method, `run()`.
+
+```java
+public static void run()
+{
+ // what your applet does
+}
+```
+
+The `run()` method contains things you may want to run in a separate thread. Such as initialization code, the previous loop for your applet, or anything else that runs in its thread. This allows you to create new objects and call methods from within `run()` while also allowing the objects to pass within the thread. 
+
+Thread execution is stopped after running the threads. This can only be accomplished by including the `stop()` method as shown below:
+
+```java
+public void stop()
+{
+if (runner != null)
+{
+runner.stop();//stops execution
+runner = null;//sets variable to value nulll
+}
+}
+```
+
+The `stop()` method in the preceding code achieves two things:
+- Disables the thread's execution.
+- Sets the variable (runner) of the thread to null - When a variable is set to null, the thread object uses its previously contained variable for garbage collection. This results in the applet being erased from memory after a predetermined amount of time. The start method sets a new thread and reloads the page. As a result, we create a thread that runs on this applet.
+
+### Program to create an applet with threads
+
+```java
+import java.applet.applet;
+import java.awt;
+public class Thread extends MyApplet() {
+    int w, h;
+    int z = 0;
+    boolean thread;
+    Thread t = null;
+public void init(){
+    system.out.println("init(): begin");
+    w = getSize().w;   // Defines the width
+    h = getSize().h;
+    setBg( color.#FFFF00 );
+    system.out.println("init(): stop");
+}
+public void destroy(){
+    system.out.println("kill()");
+}
+public void start() {
+    system.out.println("begin(): begin");
+    if ( t == null){
+        system.out.println("begin(): created thread");
+       t = new Thread();
+       system.out.println("begin(): starting thread");
+       t = false;
+       t.start();
+    }
+    else{
+        if(thread){
+            thread = true;
+            system.out.println("begin(): notifies thread");
+            synchronized(this){
+                notify();
+            }
+        }
+    }
+    system.out.println("begin(): stop");
+}
+}
+```

```

Comment:
Explain what is happening in this program - add some helpful inline comments and a view of the output will be very helpful to readers.